deps/zlib/zlib.gyp

```diff
@@ -48,6 +48,9 @@
               '.',
             ],
           },
+          'defines': [
+            'HAVE_HIDDEN',
+          ],
```

Comment:
Move this to the `'OS!="win"'` conditional below.  The `__attribute__((visibility("hidden")))` doesn't work with Visual Studio (see https://ci.nodejs.org/job/node-compile-windows/6701/label=win-vs2015/consoleFull) and it probably isn't needed anyway because we use deps/zlib/win32/zlib.def to control what is and isn't exported.

Alternatively, if you teach tools/mkssldef.py to generate UNIX linker maps, we can unify the two approaches.